### DISCUSSION TITLE
Can we use this clean architecture with Windows Form or WPF application?

### DISCUSSION THREAD

**Discussion by imranyaseennetconsent** [jasontaylordev/CleanArchitecture#752] [Q&A]
Hi,

I am going to use this clean architecture with my MVC application but I have another WPF application with different responsibilities most of which will be done while developing the MVC application. My question is can I use this architecture with the WPF application as well? I am asking this question because for clean architecture the sample applications are always used as web applications and I didn't see any windows or WPF applications so I need to confirm before using this architecture.

**Comment by nazare2020** [comment]
Hi,

As this is just a solution template, like other 'clean architect' solutions from other devs, you can remove or add the functionality you need or don't need. The WebUI project calls the api controllers (also in the WebUI project) but you could separate this into an API project and then call that from your Windows Form application.

You can add Microsoft.Extensions.DependencyInjection package to the Windows Forms application too and call MediatR commands directly too, although I haven't got that to work yet.

I would be interested to hear if you managed to get a working solution. We are just starting on migrating from a number of legacy Windows Forms apps to the web so I am looking at how I can integrate this design pattern into Windows Forms so I can take it forward to the web solution at a later date.

**Comment by RebinM49** [comment]
> Hi,
> 
> I am going to use this clean architecture with my MVC application but I have another WPF application with different responsibilities most of which will be done while developing the MVC application. My question is can I use this architecture with the WPF application as well? I am asking this question because for clean architecture the sample applications are always used as web applications and I didn't see any windows or WPF applications so I need to confirm before using this architecture.

It's also my question,if it's possible to use any other UI framework (like console.Gui ) instead of web UI .
